server/planning/events/events_spike.py

```diff
@@ -62,6 +62,52 @@ def update(self, id, updates, original):
 
         return item
 
+    def on_updated(self, updates, original):
+        # Spike associated planning
+        planning_spike_service = get_resource_service('planning_spike')
+        for planning in list(get_resource_service('planning').find(where={
+            'event_item': original[config.ID_FIELD],
+            'state': WORKFLOW_STATE.DRAFT
+        })):
+            if planning.get('state') == WORKFLOW_STATE.DRAFT:
+                planning_spike_service.patch(
+                    planning[config.ID_FIELD],
+                    {'state': 'spiked'}
+                )
+
+        # When a planning item associated with this event is spiked
+        # If there were any failures in removing assignments
+        # Send those notifications here
+
+        assignments_to_fetch = []
+        assignments_service = get_resource_service('assignments')
+        for planning in list(get_resource_service('planning').find(
```

Comment:
It might actually be better to perform a single query for `event_item` only, and use an `if planning['state'] == 'draft'/'spiked'` instead of performing two fetch/iterate cycles (i.e. merging above spike and below assignment to fetch)